Duplicate shape prevention

cornbrandon opened the pull request:
Preventing a duplicate shape from being drawn when the mousedown event fails to fire.  This occured when clicking on some map controls.

Added the _flag_new_shape boolean to help prevent the same shape being drawn multiple times.  This preserves the _ARR_LatLon values as to not affect the getAreaLatLng function.

cornbrandon committed:
Duplicate shape prevention

Preventing a duplicate shape from being drawn when the mousedown event fails to fire.  This occured when clicking on some map controls.

sandropibia commented:
Hi Crunchy005, I'll look at the provided change and come back here if I tested your work. For so far thanks for cooperating.

Greez

cornbrandon committed:
Ray Tracing for Markers

Added Ray Trace algorithm to more accurately grab the Markers and Circle markers from the selected area.  The previous solution would look within the selected area bounds and sometimes grab markers ouyside of the area.

Commit:
Update Leaflet.SelectAreaFeature.js
